From a17cc18a6214ae7782fa6894359adaa46ab4fec5 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 12 Dec 2013 18:27:09 +0800
Subject: [PATCH 49/68] ENGR00291904 Enable WFD connection on none google
 nexus device

Do not make RTP socket's connect(), if remote port is 0
Several jb4.2 WFD Source devices does not report Sink with
server_port field

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 media/libstagefright/foundation/ANetworkSession.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/foundation/ANetworkSession.cpp b/media/libstagefright/foundation/ANetworkSession.cpp
index dae1ef6..41e2502 100644
--- a/media/libstagefright/foundation/ANetworkSession.cpp
+++ b/media/libstagefright/foundation/ANetworkSession.cpp
@@ -1168,12 +1168,18 @@ status_t ANetworkSession::connectUDPSession(
         int32_t sessionID, const char *remoteHost, unsigned remotePort) {
     Mutex::Autolock autoLock(mLock);
 
+    status_t err = OK;
     ssize_t index = mSessions.indexOfKey(sessionID);
 
     if (index < 0) {
         return -ENOENT;
     }
 
+    if (remotePort <= 0) {
+        ALOGI("Remote port is %d, no need to connect!", remotePort);
+        return err;
+    }
+
     const sp<Session> session = mSessions.valueAt(index);
     int s = session->socket();
 
@@ -1182,7 +1188,6 @@ status_t ANetworkSession::connectUDPSession(
     remoteAddr.sin_family = AF_INET;
     remoteAddr.sin_port = htons(remotePort);
 
-    status_t err = OK;
     struct hostent *ent = gethostbyname(remoteHost);
     if (ent == NULL) {
         err = -h_errno;
-- 
1.8.0

